wolfcrypt/src/port/iotsafe/iotsafe.c

```diff
@@ -859,6 +881,14 @@ static int iotsafe_sign_hash(byte *privkey_idx, uint16_t id_size,
             byte sig_hdr[3];
             if (hex_to_bytes(resp, sig_hdr, 3) < 0) {
                ret = BAD_FUNC_ARG;
+#ifdef IOTSAFE_SIGN_NO_PADDING
+            } else if ((sig_hdr[0] == IOTSAFE_TAG_SIGNATURE_FIELD) &&
```

Comment:
Can we not detect the no pad case by checking the value of `sig_hdr[1] == 0`? Is no padding the right term for this? Looks like the only difference is the size is omitted.

Reply:
Padding may not be the best term for this, basically the expected case is a 16-bit length of 0x00xx, but on some platforms we get an 8-bit length.
I can't merge this with the other case if that's what you mean, as we need to use a different offset in the response for each case.

Reply:
Renamed this to `IOTSAFE_SIG_8BIT_LENGTH`, as it's an 8-bit (vs 16-bit normally) length field for a signature response.